codec-dns/src/main/java/io/netty/handler/codec/dns/AbstractDnsOptPseudoRrRecord.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.dns;
+
+import io.netty.util.internal.StringUtil;
+import io.netty.util.internal.UnstableApi;
+
+/**
+ * An <a href="https://tools.ietf.org/html/rfc6891#section-6.1">OPT RR</a> record.
+ *
+ * This is used for <a href="https://tools.ietf.org/html/rfc6891#section-6.1.3">
+ *     Extension Mechanisms for DNS (EDNS(0))</a>.
+ */
+@UnstableApi
+public abstract class AbstractDnsOptPseudoRrRecord extends AbstractDnsRecord implements DnsOptPseudoRecord {
+
+    protected AbstractDnsOptPseudoRrRecord(int maxPayladSize, int extendedRcode, int version) {
+        super(StringUtil.EMPTY_STRING, DnsRecordType.OPT, maxPayladSize, packIntoLong(extendedRcode, version));
+    }
+
+    protected AbstractDnsOptPseudoRrRecord(int maxPayladSize) {
+        super(StringUtil.EMPTY_STRING, DnsRecordType.OPT, maxPayladSize, 0);
+    }
+
+    // See https://tools.ietf.org/html/rfc6891#section-6.1.3
+    private static long packIntoLong(int val, int val2) {
+        // We are currently not support DO and Z fields, just use 0.
+        return ((val & 0xff) << 24 | (val2 & 0xff) << 16 | (0 & 0xff) <<  8 | 0 & 0xff) & 0xFFFFFFFFL;
```

Comment:
I would prefer to keep it as it is to make it "easier" to add support later. Its not a big deal imho to keep one more shift.